### PR TITLE
Feature/eui 193

### PR DESCRIPTION
**Pull request by udaydenduluri33**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Comment by udaydenduluri33** [comment]
EUI-193

File: src/app/containers/app/app.component.ts [hunk 1]
```diff
@@ -9,13 +9,23 @@ import { Store } from '@ngrx/store';
...
-export class AppComponent {
+export class AppComponent implements OnInit {
   constructor(
     private logger: LoggerService,
     private store: Store<fromActions.State>
   ) {
   }
 
+  ngOnInit() {
+    this.logger.trace('AppComponent: logger.trace()');
```
Comment (fazanki) [inline src/app/containers/app/app.component.ts]: this should be moved from here to <exui-logger> and component name pass as an input property

File: src/app/services/logger/actions/logger-state.actions.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import { Action } from '@ngrx/store';
```
Comment (fazanki) [inline src/app/services/logger/actions/logger-state.actions.ts]: the file needs to be moved to app/store dir

File: src/app/services/logger/effects/logger.effects.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@angular/core';
```
Comment (fazanki) [inline src/app/services/logger/effects/logger.effects.ts]: this needs to be moved to app/store dir

File: src/app/services/logger/exui-logger/exui-logger.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import { TestBed, async } from '@angular/core/testing';
```
Comment (fazanki) [inline src/app/services/logger/exui-logger/exui-logger.component.spec.ts]: this needs to be moved to app/component/exu-logger

File: src/app/services/logger/exui-logger/exui-logger.component.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
+import {Component, OnInit, OnDestroy} from '@angular/core';
```
Comment (fazanki) [inline src/app/services/logger/exui-logger/exui-logger.component.ts]: this needs to be moved to app/component/exu-logger

File: src/app/services/logger/exui-logger/exui-logger.component.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+  debugSubscribtion: any;
+  infoSubscribtion: any;
+  warningSubscribtion: any;
+  errorSubscribtion: any;
+  fatalSubscribtion: any;
+  traceSubscribtion: any;
+
+  constructor(private ngxLogger: NGXLogger, private store: Store<LoggerMessage>) { }
+  ngOnInit(): void {
```
Comment (fazanki) [inline src/app/services/logger/exui-logger/exui-logger.component.ts]: here needs to come code from app.component ngOnInit and the component name needs to be dynamic

File: src/app/services/logger/logger.module.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import {NgModule, CUSTOM_ELEMENTS_SCHEMA} from '@angular/core';
```
Comment (fazanki) [inline src/app/services/logger/logger.module.ts]: this needs to be removed and the ExuiLoggerComponent needs to be added to apps shared.module

File: src/app/services/logger/reducers/index.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
+import {
```
Comment (fazanki) [inline src/app/services/logger/reducers/index.ts]: needs to go to the store dir

File: src/app/services/logger/reducers/logger.reducer.spec.ts [hunk 9]
```diff
@@ -0,0 +1,85 @@
+import {loggerReducer, initialState} from './logger.reducer';
```
Comment (fazanki) [inline src/app/services/logger/reducers/logger.reducer.spec.ts]: file needs to go to the store dir

File: src/app/services/logger/reducers/logger.reducer.ts [hunk 10]
```diff
@@ -0,0 +1,63 @@
+import { LoggerStateAction, LoggerStateActionTypes } from '../actions/logger-state.actions';
+
```
Comment (fazanki) [inline src/app/services/logger/reducers/logger.reducer.ts]: file needs to go to the store dir

File: src/app/services/logger/selectors/logger.selector.spec.ts [hunk 11]
```diff
@@ -0,0 +1,113 @@
+import { TestBed } from '@angular/core/testing';
```
Comment (fazanki) [inline src/app/services/logger/selectors/logger.selector.spec.ts]: needs to move to the store dir

File: src/app/services/logger/selectors/logger.selector.ts [hunk 12]
```diff
@@ -0,0 +1,32 @@
+import {createSelector} from '@ngrx/store';
```
Comment (fazanki) [inline src/app/services/logger/selectors/logger.selector.ts]: move to the store dir

File: src/app/services/logger/exui-logger/exui-logger.component.spec.ts (hunk 4 above)
Comment (smogese) [reply src/app/services/logger/exui-logger/exui-logger.component.spec.ts]: Yes, why is the component test in the services?

File: src/app/containers/app/app.component.ts (hunk 1 above)
Comment (udaydenduluri33) [reply src/app/containers/app/app.component.ts]: Done

File: src/app/services/logger/effects/logger.effects.ts (hunk 3 above)
Comment (udaydenduluri33) [reply src/app/services/logger/effects/logger.effects.ts]: Done

File: src/app/services/logger/actions/logger-state.actions.ts (hunk 2 above)
Comment (udaydenduluri33) [reply src/app/services/logger/actions/logger-state.actions.ts]: Done

File: src/app/services/logger/exui-logger/exui-logger.component.spec.ts (hunk 4 above)
Comment (udaydenduluri33) [reply src/app/services/logger/exui-logger/exui-logger.component.spec.ts]: Done

File: src/app/services/logger/exui-logger/exui-logger.component.ts (hunk 5 above)
Comment (udaydenduluri33) [reply src/app/services/logger/exui-logger/exui-logger.component.ts]: Done

File: src/app/services/logger/reducers/index.ts (hunk 8 above)
Comment (udaydenduluri33) [reply src/app/services/logger/reducers/index.ts]: There is already an index.ts in the store directory

File: src/app/services/logger/reducers/logger.reducer.spec.ts (hunk 9 above)
Comment (udaydenduluri33) [reply src/app/services/logger/reducers/logger.reducer.spec.ts]: Done

File: src/app/services/logger/reducers/logger.reducer.ts (hunk 10 above)
Comment (udaydenduluri33) [reply src/app/services/logger/reducers/logger.reducer.ts]: Done

File: src/app/services/logger/selectors/logger.selector.spec.ts (hunk 11 above)
Comment (udaydenduluri33) [reply src/app/services/logger/selectors/logger.selector.spec.ts]: Done

File: src/app/services/logger/selectors/logger.selector.ts (hunk 12 above)
Comment (udaydenduluri33) [reply src/app/services/logger/selectors/logger.selector.ts]: Done

File: src/app/services/logger/reducers/index.ts (hunk 8 above)
Comment (udaydenduluri33) [reply src/app/services/logger/reducers/index.ts]: DOne

File: src/app/services/logger/logger.module.ts (hunk 7 above)
Comment (udaydenduluri33) [reply src/app/services/logger/logger.module.ts]: Done. Thanks

**Pull request merged**